paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -222,66 +222,53 @@ internal class CustomerSheetViewModel @Inject constructor(
 
                 savedPaymentSelection = paymentSelection
 
-                val isGooglePayReady = googlePayRepositoryFactory(
-                    if (isLiveModeProvider()) {
-                        GooglePayEnvironment.Production
-                    } else {
-                        GooglePayEnvironment.Test
+                val isGooglePayReadyAndEnabled = isGooglePayReadyAndEnabled()
+
+                val selectState = CustomerSheetViewState.SelectPaymentMethod(
+                    title = configuration.headerTextForSelectionScreen,
+                    savedPaymentMethods = paymentMethods,
+                    paymentSelection = paymentSelection,
+                    isLiveMode = isLiveModeProvider(),
+                    isProcessing = false,
+                    isEditing = false,
+                    isGooglePayEnabled = isGooglePayReadyAndEnabled(),
+                    primaryButtonVisible = false,
+                    // TODO (jameswoo) translate
+                    primaryButtonLabel = "Confirm",
+                    errorMessage = null,
+                )
+
+                if (paymentMethods.isEmpty() && !isGooglePayReadyAndEnabled) {
+                    val paymentMethodCode = PaymentMethod.Type.Card.code
+                    val observe = buildFormObserver(paymentMethodCode = paymentMethodCode)
+
+                    backStack.update {
+                        listOf(
+                            selectState,
```

Comment:
Do we really want to allow the user to go back to the empty payment methods screen, or should the add screen be the only screen on the back stack?